vault/logical_system_paths.go

```diff
@@ -64,38 +84,208 @@ func (b *SystemBackend) configPaths() []*framework.Path {
 		{
 			Pattern: "config/ui/headers/$",
 
-			Callbacks: map[logical.Operation]framework.OperationFunc{
-				logical.ListOperation: b.handleConfigUIHeadersList,
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.ListOperation: &framework.PathOperation{
+					Callback: b.handleConfigUIHeadersList,
+					Summary:  "Return a list of configured UI headers.",
+				},
 			},
 
 			HelpDescription: strings.TrimSpace(sysHelp["config/ui/headers"][0]),
 			HelpSynopsis:    strings.TrimSpace(sysHelp["config/ui/headers"][1]),
 		},
 
 		{
-			Pattern:         "generate-root(/attempt)?$",
+			Pattern: "generate-root(/attempt)?$",
+			Fields: map[string]*framework.FieldSchema{
+				"otp": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Description: "Specifies a base64-encoded 16-byte value.",
+				},
+				"pgp_key": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Description: "Specifies a base64-encoded PGP public key.",
+				},
+			},
+			Operations: map[logical.Operation]framework.OperationHandler{
+				logical.ReadOperation: &framework.PathOperation{
+					Summary: "Read the configuration and progress of the current root generation attempt.",
+				},
+				logical.UpdateOperation: &framework.PathOperation{
+					Summary:     "Initializes a new root generation attempt.",
+					Description: "Only a single root generation attempt can take place at a time. One (and only one) of otp or pgp_key are required.",
+				},
+				logical.DeleteOperation: &framework.PathOperation{
+					Summary: "Cancels any in-progress root generation attempt.",
+				},
+			},
+
+			HelpSynopsis:    strings.TrimSpace(sysHelp["generate-root"][0]),
+			HelpDescription: strings.TrimSpace(sysHelp["generate-root"][1]),
+		},
+		{
```

Comment:
I was about to comment that we need to add `generate-root/update` and saw this. 👍